### PR TITLE
Tests for SystemRegister 

### PR DESCRIPTION
**Pull request by simen-rekkedal**
Moved SystemUserService Mocklist to the proper place as a SystemUserServiceMock,
wrote a proper SystemUserService class to call the Repository

Started writing Automatic tests for SystemRegister Controller and Db

More to come in the same theme, but trying to keep the PR smallish.

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: test/Altinn.Platform.Authentication.Tests/Controllers/SystemRegisterControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        [Fact]
+        public async Task SystemRegister_Get_ListofAll()
+        {
+            HttpClient client = GetTestClient(_sblCookieDecryptionService.Object, _userProfileService.Object);            
+            HttpRequestMessage request = new(HttpMethod.Get, $"/authentication/api/v1/systemregister");
+            HttpResponseMessage response = await client.SendAsync(request, HttpCompletionOption.ResponseContentRead );
+            List<RegisteredSystem> list = JsonSerializer.Deserialize<List<RegisteredSystem>>(await response.Content.ReadAsStringAsync(), jsonOptions);
+            Assert.Equal(2, list.Count);
```
Comment (acn-dgopa) [inline test/Altinn.Platform.Authentication.Tests/Controllers/SystemRegisterControllerTests.cs:54]: I'm trying to understand this assert. Im not seeing any mock data setup for the systemregister. I see for userprofileservice. So how does this list returns your expected value of 2?

File: test/Altinn.Platform.Authentication.Tests/Mocks/SystemUserServiceMock.cs [hunk 2]
```diff
@@ -57,7 +86,61 @@ public Task<int> SetDeleteFlagOnSystemUser(Guid systemUserId)
...
+            theMockList[array].IntegrationTitle = request.IntegrationTitle;
+            theMockList[array].ProductName = request.ProductName;
+            return Task.FromResult(1);
+        }
+
+        /// <summary>
+        /// Helper method during development, just some Mock data.
+        /// </summary>
+        /// <returns></returns>        
+        private static List<SystemUser> MockDataHelper()
```
Comment (acn-dgopa) [inline test/Altinn.Platform.Authentication.Tests/Mocks/SystemUserServiceMock.cs:99]: Is there any advantage og hardcoding it as a method over using a json file for the mockdata? If json file, probably it is more readable and data kept separately

File: src/Authentication/Services/SystemUserService.cs [hunk 3]
```diff
@@ -14,133 +15,94 @@ namespace Altinn.Platform.Authentication.Services
...
+                return 1;
+            }
+
+            return 0;
         }
 
         /// <summary>
         /// Replaces the values for the existing system user with those from the update 
         /// </summary>
         /// <returns></returns>
```
Comment (acn-dgopa) [inline src/Authentication/Services/SystemUserService.cs:91]: you have probably defined the summary,params, returns in the ISystemUserService?

File: src/Authentication/Services/SystemUserService.cs [hunk 4]
```diff
@@ -14,133 +15,94 @@ namespace Altinn.Platform.Authentication.Services
...
+            SystemUser search = await _repository.GetSystemUserById(systemUserId);
             
-            return Task.FromResult(search);
+            return search;
         }
 
         /// <summary>
         /// Set the Delete flag on the identified SystemUser
         /// </summary>
         /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Authentication/Services/SystemUserService.cs:75]: ok, changed in the next PR

File: src/Core/RepositoryInterfaces/ISystemUserRepository.cs [hunk 5]
```diff
@@ -33,4 +33,12 @@ public interface ISystemUserRepository
     /// </summary>
     /// <param name="id">The Guid id</param>
     Task SetDeleteSystemUserById(Guid id);
+
+    /// <summary>
+    /// Updates the Product Name on an Integration by Guid
+    /// </summary>
+    /// <param name="guid"></param>
+    /// <param name="productName"></param>
+    /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Core/RepositoryInterfaces/ISystemUserRepository.cs:42]: ok

File: src/Persistance/RepositoryImplementations/SystemUserRepository.cs [hunk 6]
```diff
@@ -168,6 +168,38 @@ public async Task<Guid> InsertSystemUser(SystemUser toBeInserted)
...
+            command.Parameters.AddWithValue("id", guid);
+            command.Parameters.AddWithValue("product_name", productName);
+
+            return await command.ExecuteEnumerableAsync()
+                .SelectAwait(ConvertFromReaderToInt)
+                .FirstOrDefaultAsync();
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Authentication // SystemRegisterRepository // InsertSystemUser // Exception");
```
[the remark this answers is not in the document]
Comment (simen-rekkedal) [reply src/Persistance/RepositoryImplementations/SystemUserRepository.cs:193]: ok

File: test/Altinn.Platform.Authentication.Tests/Controllers/SystemRegisterControllerTests.cs (hunk 1 above)
Comment (simen-rekkedal) [reply test/Altinn.Platform.Authentication.Tests/Controllers/SystemRegisterControllerTests.cs:54]: there is a Mock service which returns a list of 4, changed the Assert to be greater than 0.

File: test/Altinn.Platform.Authentication.Tests/Mocks/SystemUserServiceMock.cs (hunk 2 above)
Comment (simen-rekkedal) [reply test/Altinn.Platform.Authentication.Tests/Mocks/SystemUserServiceMock.cs:99]: will expand to json file in future PR

File: src/Authentication/Services/SystemUserService.cs (hunk 3 above)
Comment (simen-rekkedal) [reply src/Authentication/Services/SystemUserService.cs:91]: ok

**Pull request merged**